assignment2.py

```diff
@@ -0,0 +1,155 @@
+class ListNode:
+    def __init__(self, value):
+        self.next = None
+        self.value = value
+        
+def string_compression(s):
+    count = 1
+
+    compressed = []
+
+    for i in range(1, len(s)):
+        if s[i - 1] == s[i] :
+            count += 1
+
+        else:
+            compressed.append(s[i - 1])
+            compressed.append(str(count))
```

Comment:
Because you're calling `join()` on this anyways, you might as well do this in a single string addition instead of two separate calls to `append`.
```suggestion
            compressed.append(s[i - 1] + str(count))
```